### PR TITLE
Fix pydm installation

### PR DESCRIPTION
**Pull request by fernandohds564**
`pydm` was being installed via setup, using preferably the `deploy_tag` variable. This results in error since we are not tagging our `pydm` fork repo anymore. Instead of just defining a version for installation, I've decided to moved `pydm` and `timechart` to the pip installation instead.

Maybe It would be better to move these packages to the python role instead, since they are not only related to our `sirius-apps` packages. **What do you guys think?**

### CODE REVIEW

**Comment by lerwys** [comment]
I have some questions:

 - Why are we not tagging our pydm fork anymore?
 - If we are installing via pip we are not installing our fork, but the upstream pydm version I think. Is that what we want?
 - In my opinion roles should be generic and just do one specific thing (our sirius-apps and desktop-apps are not good examples of this...). Making the python role installing pydm and timechart would make it very specific to our use case.

**Comment by lerwys** [comment]
Also, I think we should avoid committing commented code. It makes reading the diffs more difficult, it confuses other developer that see commented code and it's easy for the comment to become out-of-date. We  can use `git diff` and `git log --folow --  <filename>` to see all commits that changed a file, if we need to rollback or check how things were done before.

**Comment by fernandohds564** [comment]
- We are not tagging it because we decided to use the upstream version. The Controls Group already use this version and we wanted to standardize the versions. Since the master branch of our fork is identical to the master version of the upstream, this is not a problem for us for now. If in the future we decide to use a different functionality only present in our pydm, we can change back the installation in ansible.
 - already answered :smiley: 
 - I agree with you, I think our python role should be reformulated. However this is not the scope of this PR and looking things how they are today, I thought the python role, or even the epics role, could be more appropriate, given that `pydm` and `timechart`, just like `cs-studio`, have a more general scope than our sirius apps codes. But I don't know if it's better to move them, that's why I asked you guys about it...

**Comment by fernandohds564** [comment]
Ok! I will remove the comments!
I also don't like them, but @xresende does :grin:, that's  why I kept them.

**Comment by xresende** [comment]
I sometimes do leave comments. __my__ commented code!  :laughing:

**Comment by lerwys** [comment]
@fernandohds564 Nice nice. Ok then.

About the roles. Yes, in the near future we can maybe split the python role into lnls-ans-role-python and lnls-ans-role-python-dev. But what about converting pydm into a role (just like cs-studio) having the python role as a dependency? THen all of the environment variables and related packages could be inside that role.

**Comment by fernandohds564** [comment]
Ok @lerwys! good idea! I will do it!

**Comment by lerwys** [comment]
Nice!

**Comment by fernandohds564** [comment]
@lerwys, @anacso17, @xresende,  I've implemented the pydm role. Could you take a look to see if it's ok?

File: roles/lnls-ans-role-pydm/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+galaxy_info:
+  author: Lucas Russo
```
Comment (xresende) [inline roles/lnls-ans-role-pydm/meta/main.yml]: (@fernandohds564 , @lerwys)
@fernandohds564 , I think you could become author (main responsable) for this role, even though it uses a previous @lerwys role as model/template. Is this standard, @lerwys ?

File: roles/lnls-ans-role-pydm/molecule/default/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+verifier:
+  name: testinfra
+  options:
+    verbose: true
+  lint:
+    name: flake8
+driver:
+  name: docker
+platforms:
+  - name: lnls-ans-role-sirius-apps-instance
```
Comment (xresende) [inline roles/lnls-ans-role-pydm/molecule/default/molecule.yml]: @fernandohds564 , should `lnls-ans-role-sirius-apps-instance` be changed here?

Comment (fernandohds564) [reply roles/lnls-ans-role-pydm/molecule/default/molecule.yml]: I don't know @xresende. Should I, @lerwys ?

File: roles/lnls-ans-role-pydm/meta/main.yml (hunk 1 above)
Comment (lerwys) [reply roles/lnls-ans-role-pydm/meta/main.yml]: Sure. You can put yourself as the author. The author name here is mainly (I think) used by https://galaxy.ansible.com/ (we are not using it), but it serves as a documentation as well.

File: roles/lnls-ans-role-pydm/molecule/default/molecule.yml (hunk 2 above)
Comment (lerwys) [reply roles/lnls-ans-role-pydm/molecule/default/molecule.yml]: Yes, let's change it. Any name would work, but this name is printed in the test logs (travis), so it's better to use a more descriptive name

**Comment by xresende** [comment]
@lerwys, soon I will create a PR that modifies all roles so that they use the package version constants. Then I will implement the syntax you suggested above that allows for using the latest versions of the packages.

**Comment by lerwys** [comment]
@xresende Ok then. Great!

**Comment by fernandohds564** [comment]
hi @lerwys ! thanks for the review! I will make the changes you suggested!

File: roles/lnls-ans-role-pydm/meta/main.yml (hunk 1 above)
Comment (fernandohds564) [reply roles/lnls-ans-role-pydm/meta/main.yml]: done!

File: roles/lnls-ans-role-pydm/molecule/default/molecule.yml (hunk 2 above)
Comment (fernandohds564) [reply roles/lnls-ans-role-pydm/molecule/default/molecule.yml]: done!

**Comment by fernandohds564** [comment]
Done @lerwys!

@xresende and @lerwys, could you take another look?

**Comment by lerwys** [comment]
@fernandohds564 Seems good! I would just inverted the order of tasks `Ensure PyDM environment file is installed` and `Install PyDM`, just because the template files only make sense if pydm is already installed. No need to add conditions, I think. If pydm fails for some reason ansible will stop the tasks.

**Comment by lerwys** [comment]
One more thing @fernandohds564 . I would advise against merging master into feature branches (https://github.com/lnls-sirius/lnls-ansible/pull/7/commits/f266c4761a28e20941cddec896ffbf058a2555c1). They are difficult to revert after merging the feature branch into master.

**Comment by fernandohds564** [comment]
Ok @lerwys, sorry about that!

**Comment by fernandohds564** [comment]
> @fernandohds564 Seems good! I would just inverted the order of tasks `Ensure PyDM environment file is installed` and `Install PyDM`, just because the template files only make sense if pydm is already installed. No need to add conditions, I think. If pydm fails for some reason ansible will stop the tasks.

done!

**Comment by lerwys** [comment]
@fernandohds564 All good to me!

**Pull request merged**